logproof/src/math.rs

```diff
@@ -92,14 +92,20 @@ pub trait InfinityNorm {
 
 impl<R> InfinityNorm for Polynomial<R>
 where
-    R: Ring + Ord,
+    R: Ring + Ord + Clone,
 {
     type Output = R;
 
     fn infinity_norm(&self) -> Self::Output {
-        self.coeffs
-            .iter()
-            .fold(R::zero(), |max, x| if x > &max { x.clone() } else { max })
+        self.coeffs.iter().fold(R::zero(), |max, x| {
+            let x_neg = x.clone().neg();
+            let x_abs = x.min(&x_neg);
```

Comment:
Should this be max?

Reply:
I guess it depends on your intent, and more technically it depends on the ring's `Ord` impl. Because of our 2s complement encoding in logproof, this number is more relevant to us. (It matches what we've [done here](https://github.com/Sunscreen-tech/Sunscreen/blob/4db46f0967c8862c55928cf46a8a14ce4a76861b/logproof/src/linear_relation.rs#L1145) anyways).

Reply:
Sorry, I mispoke, I'm nearing brain dead over here. The above is generally correct for all of our rings which defer their ord instance to their underlying unsigned integer representation. Valid values are in the range `[0..R::MODULUS]` with `R::MODULUS/2` being the negative cutoff. So the absolute value of `x` will always be the minimum of `x, x.neg()` as it one will be in the first half of the range, and the other in the greater half of the range.